### PR TITLE
JDK-8301495: Replace NULL with nullptr in cpu/ppc

### PR DESCRIPTION
**Pull request by jdksjolen**
Hi, this PR changes all occurrences of NULL to nullptr for the subdirectory cpu/ppc. Unfortunately the script that does the change isn't perfect, and so we
need to comb through these manually to make sure nothing has gone wrong. I also review these changes but things slip past my eyes sometimes.

Here are some typical things to look out for:

1. No changes but copyright header changed (probably because I reverted some changes but forgot the copyright).
2. Macros having their NULL changed to nullptr, these are added to the script when I find them. They should be NULL.
3. nullptr in comments and logs. We try to use lower case "null" in these cases as it reads better. An exception is made when code expressions are in a comment.

An example of this:

```c++
// This function returns null
void* ret_null();
// This function returns true if *x == nullptr
bool is_nullptr(void** x);
```

Note how `nullptr` participates in a code expression here, we really are talking about the specific value `nullptr`.

Thanks!

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301495](https://bugs.openjdk.org/browse/JDK-8301495): Replace NULL with nullptr in cpu/ppc

### Reviewers
 * [Richard Reingruber](https://openjdk.org/census#rrich) (@reinrich - **Reviewer**)
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**)
 * [Tyler Steele](https://openjdk.org/census#tsteele) (@backwaterred - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/12323/head:pull/12323` \
`$ git checkout pull/12323`

Update a local copy of the PR: \
`$ git checkout pull/12323` \
`$ git pull https://git.openjdk.org/jdk.git pull/12323/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12323`

View PR using the GUI difftool: \
`$ git pr show -t 12323`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12323.diff">https://git.openjdk.org/jdk/pull/12323.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/12323#issuecomment-1469990478)

### CODE REVIEW

File: src/hotspot/cpu/ppc/abstractInterpreter_ppc.cpp [hunk 1]
```diff
@@ -94,9 +94,9 @@ int AbstractInterpreter::size_activation(int max_stack,
 //
 // Parameters:
 //
-// interpreter_frame != NULL:
+// interpreter_frame != null:
```
Comment (reinrich) [inline src/hotspot/cpu/ppc/abstractInterpreter_ppc.cpp:97]:
```suggestion
// interpreter_frame != nullptr:
```

File: src/hotspot/cpu/ppc/frame_ppc.cpp [hunk 2]
```diff
@@ -111,7 +111,7 @@ bool frame::safe_for_sender(JavaThread *thread) {
     }
 
     // At this point, there still is a chance that fp_safe is false.
-    // In particular, (fp == NULL) might be true. So let's check and
+    // In particular, (fp == null) might be true. So let's check and
```
Comment (reinrich) [inline src/hotspot/cpu/ppc/frame_ppc.cpp:114]:
```suggestion
    // In particular, (fp == nullptr) might be true. So let's check and
```

File: src/hotspot/cpu/ppc/interp_masm_ppc_64.cpp [hunk 3]
```diff
@@ -917,7 +917,7 @@ void InterpreterMacroAssembler::lock_object(Register monitor, Register object) {
     //   // We stored the monitor address into the object's mark word.
     // } else if (THREAD->is_lock_owned((address)displaced_header))
     //   // Simple recursive case.
-    //   monitor->lock()->set_displaced_header(NULL);
+    //   monitor->lock()->set_displaced_header(null);
```
Comment (reinrich) [inline src/hotspot/cpu/ppc/interp_masm_ppc_64.cpp:920]:
```suggestion
    //   monitor->lock()->set_displaced_header(nullptr);
```

File: src/hotspot/cpu/ppc/interp_masm_ppc_64.cpp [hunk 4]
```diff
@@ -978,7 +978,7 @@ void InterpreterMacroAssembler::lock_object(Register monitor, Register object) {
 
     // } else if (THREAD->is_lock_owned((address)displaced_header))
     //   // Simple recursive case.
-    //   monitor->lock()->set_displaced_header(NULL);
+    //   monitor->lock()->set_displaced_header(null);
```
Comment (reinrich) [inline src/hotspot/cpu/ppc/interp_masm_ppc_64.cpp:981]:
```suggestion
    //   monitor->lock()->set_displaced_header(nullptr);
```

File: src/hotspot/cpu/ppc/interp_masm_ppc_64.cpp [hunk 5]
```diff
@@ -1028,12 +1028,12 @@ void InterpreterMacroAssembler::unlock_object(Register monitor) {
 
     // template code:
     //
-    // if ((displaced_header = monitor->displaced_header()) == NULL) {
-    //   // Recursive unlock. Mark the monitor unlocked by setting the object field to NULL.
-    //   monitor->set_obj(NULL);
+    // if ((displaced_header = monitor->displaced_header()) == null) {
```
Comment (reinrich) [inline src/hotspot/cpu/ppc/interp_masm_ppc_64.cpp:1031]:
```suggestion
    // if ((displaced_header = monitor->displaced_header()) == nullptr) {
```

File: src/hotspot/cpu/ppc/interp_masm_ppc_64.cpp [hunk 6]
```diff
@@ -1028,12 +1028,12 @@ void InterpreterMacroAssembler::unlock_object(Register monitor) {
...
+    //   // Recursive unlock. Mark the monitor unlocked by setting the object field to null.
+    //   monitor->set_obj(null);
```
Comment (reinrich) [inline src/hotspot/cpu/ppc/interp_masm_ppc_64.cpp:1033]:
```suggestion
    //   monitor->set_obj(nullptr);
```

File: src/hotspot/cpu/ppc/interp_masm_ppc_64.cpp [hunk 7]
```diff
@@ -1028,12 +1028,12 @@ void InterpreterMacroAssembler::unlock_object(Register monitor) {
...
     // } else if (Atomic::cmpxchg(obj->mark_addr(), monitor, displaced_header) == monitor) {
     //   // We swapped the unlocked mark in displaced_header into the object's mark word.
-    //   monitor->set_obj(NULL);
+    //   monitor->set_obj(null);
```
Comment (reinrich) [inline src/hotspot/cpu/ppc/interp_masm_ppc_64.cpp:1036]:
```suggestion
    //   monitor->set_obj(nullptr);
```

File: src/hotspot/cpu/ppc/interp_masm_ppc_64.cpp [hunk 8]
```diff
@@ -1059,7 +1059,7 @@ void InterpreterMacroAssembler::unlock_object(Register monitor) {
 
     // } else if (Atomic::cmpxchg(obj->mark_addr(), monitor, displaced_header) == monitor) {
     //   // We swapped the unlocked mark in displaced_header into the object's mark word.
-    //   monitor->set_obj(NULL);
+    //   monitor->set_obj(null);
```
Comment (reinrich) [inline src/hotspot/cpu/ppc/interp_masm_ppc_64.cpp:1062]:
```suggestion
    //   monitor->set_obj(nullptr);
```

File: src/hotspot/cpu/ppc/interp_masm_ppc_64.cpp [hunk 9]
```diff
@@ -1094,7 +1094,7 @@ void InterpreterMacroAssembler::unlock_object(Register monitor) {
     Label done;
     b(done); // Monitor register may be overwritten! Runtime has already freed the slot.
 
-    // Exchange worked, do monitor->set_obj(NULL);
+    // Exchange worked, do monitor->set_obj(null);
```
Comment (reinrich) [inline src/hotspot/cpu/ppc/interp_masm_ppc_64.cpp:1097]:
```suggestion
    // Exchange worked, do monitor->set_obj(nullptr);
```

File: src/hotspot/cpu/ppc/interpreterRT_ppc.cpp [hunk 10]
```diff
@@ -99,10 +99,10 @@ void InterpreterRuntime::SignatureHandlerGenerator::pass_object() {
   Register r = jni_arg.is_register() ? jni_arg.as_register() : R11_scratch1;
 
   // The handle for a receiver will never be null.
-  bool do_NULL_check = offset() != 0 || is_static();
+  bool do_nullptr_check = offset() != 0 || is_static();
```
Comment (reinrich) [inline src/hotspot/cpu/ppc/interpreterRT_ppc.cpp:102]:
```suggestion
  bool do_null_check = offset() != 0 || is_static();
```

File: src/hotspot/cpu/ppc/interpreterRT_ppc.cpp [hunk 11]
```diff
@@ -99,10 +99,10 @@ void InterpreterRuntime::SignatureHandlerGenerator::pass_object() {
...
 
   Label do_null;
-  if (do_NULL_check) {
+  if (do_nullptr_check) {
```
Comment (reinrich) [inline src/hotspot/cpu/ppc/interpreterRT_ppc.cpp:105]:
```suggestion
  if (do_null_check) {
```

File: src/hotspot/cpu/ppc/stubGenerator_ppc.cpp [hunk 12]
```diff
@@ -2412,15 +2412,15 @@ class StubGenerator: public StubCodeGenerator {
...
     // (4) length  must not be negative.
-    // (5) src klass and dst klass should be the same and not NULL.
+    // (5) src klass and dst klass should be the same and not null.
     // (6) src and dst should be arrays.
     // (7) src_pos + length must not exceed length of src.
     // (8) dst_pos + length must not exceed length of dst.
     BLOCK_COMMENT("arraycopy initial argument checks");
 
-    __ cmpdi(CCR1, src, 0);      // if (src == NULL) return -1;
+    __ cmpdi(CCR1, src, 0);      // if (src == null) return -1;
```
Comment (reinrich) [inline src/hotspot/cpu/ppc/stubGenerator_ppc.cpp:2421]:
```suggestion
    __ cmpdi(CCR1, src, 0);      // if (src == nullptr) return -1;
```

File: src/hotspot/cpu/ppc/stubGenerator_ppc.cpp [hunk 13]
```diff
@@ -2412,15 +2412,15 @@ class StubGenerator: public StubCodeGenerator {
...
     __ extsw_(src_pos, src_pos); // if (src_pos < 0) return -1;
-    __ cmpdi(CCR5, dst, 0);      // if (dst == NULL) return -1;
+    __ cmpdi(CCR5, dst, 0);      // if (dst == null) return -1;
```
Comment (reinrich) [inline src/hotspot/cpu/ppc/stubGenerator_ppc.cpp:2423]:
```suggestion
    __ cmpdi(CCR5, dst, 0);      // if (dst == nullptr) return -1;
```

File: src/hotspot/cpu/ppc/templateInterpreterGenerator_ppc.cpp [hunk 14]
```diff
@@ -522,7 +522,7 @@ address TemplateInterpreterGenerator::generate_Reference_get_entry(void) {
   // If the receiver is null then it is OK to jump to the slow path.
   __ ld(R3_RET, Interpreter::stackElementSize, R15_esp); // get receiver
 
-  // Check if receiver == NULL and go the slow path.
+  // Check if receiver == null and go the slow path.
```
Comment (reinrich) [inline src/hotspot/cpu/ppc/templateInterpreterGenerator_ppc.cpp:525]:
```suggestion
  // Check if receiver == nullptr and go the slow path.
```

**Review by reinrich — CHANGES_REQUESTED** [review]
Looks good, just a few changes I'd like to suggest.

Thanks, Richard.

File: src/hotspot/os/posix/os_posix.cpp [hunk 15]
```diff
@@ -286,7 +286,11 @@ static char* reserve_mmapped_memory(size_t bytes, char* requested_addr) {
                        flags, -1, 0);
 
   if (addr != MAP_FAILED) {
-    MemTracker::record_virtual_memory_reserve((address)addr, bytes, CALLER_PC);
+    MemTracker::record_virtual_memory_reserve(
+        (address)addr, bytes,
+        ((MemTracker ::tracking_level() == NMT_detail)
+             ? NativeCallStack(1)
+             : NativeCallStack(NativeCallStack ::FakeMarker ::its_fake)));
```
Comment (jdksjolen) [inline src/hotspot/os/posix/os_posix.cpp:293]: Accidentally committed this, will revert to original.

**Review by jdksjolen — COMMENTED** [review]
Found a few minor issues.

**Comment by jdksjolen** [comment]
Thanks @reinrich for your review. I've fixed the issues you found and fixed some that I found also, we will see what GHA says about this commit.

**Comment by jdksjolen** [comment]
@backwaterred , would you like to check that this builds and tests for PPC :)?

**Review by reinrich — APPROVED** [review]
I've built fastdebug and release on ppc64le successfully.
Hotspot tier1 tests succeeded also.
Thanks, Richard.

**Review by TheRealMDoerr — APPROVED** [review]
LGTM.

**Review by backwaterred — APPROVED** [review]
Thanks for reaching out. This looks good to me, and builds on AIX. Testing is strange on AIX atm (there are some unrelated failures), so I'm having trouble getting a good clean run. I'm happy to fix any issues that come up later.

**Comment by jdksjolen** [comment]
Thanks for all of the help with this PR.

/integrate

**Pull request closed without merging**